src/prototype_creep_startup_tasks.js

```diff
@@ -125,18 +125,13 @@ Creep.repairStructure = function(creep) {
 };
 
 Creep.prototype.getEnergyFromHostileStructures = function() {
-  if (this.carry.energy) {
-    return false;
-  }
```

Comment:
I would keep it like this. If the creep has energy we spare a `find()` call = cpu usage